scripts/validation/embedded-examples-validation.js

```diff
@@ -0,0 +1,279 @@
+        for (const key in update.example) {
+          baseDoc[key] = update.example[key];
+        }
+      } else {
+        const results = JSONPath({ path: update.json_path, json: baseDoc, resultType: 'all' });
+
+        console.log(`\nProcessing update for '${update.name}' at path '${update.json_path}'`);
+
+        const pathParts = update.json_path.split('.');
+        const targetKey = pathParts[pathParts.length - 1];
+
+        // Check if the top-level key of the example JSON matches the target key
+        let dataToMerge = update.example;
+        if (dataToMerge.hasOwnProperty(targetKey)) {
+          dataToMerge = dataToMerge[targetKey];
+        }
+
+        if (results.length === 0) {
+          console.log(`\nPath not found, creating path: '${update.json_path}'`);
+          setValueByPath(baseDoc, update.json_path, dataToMerge); // Create the path if it doesn't exist
+        } else {
+          results.forEach(result => {
+            const parent = result.parent;
+            const parentProperty = result.parentProperty;
+            console.log(`\nMerging data at path: '${update.json_path}'`);
+            if (Array.isArray(parent[parentProperty])) {
+              // If the existing data is an array, add the update data as an array item
+              parent[parentProperty].push(dataToMerge);
+            } else {
+              // Otherwise, deep merge the existing data with the new data
+              parent[parentProperty] = deepMerge(parent[parentProperty], dataToMerge);
+            }
+          });
+        }
+      }
+    } catch (e) {
+      console.error(`\nError processing update for '${update.name}' at path '${update.json_path}'`, e);
+    }
+  });
+  fs.writeFileSync(outputPath, JSON.stringify(baseDoc, null, 2), 'utf8');
+}
+
+// Iterate over the combinedData array and apply updates for each item
+const outputDir = path.join(__dirname, 'updated-docs');
+const validationPromises = []; // Array to store all validation promises
+combinedData.forEach((item) => {
+  const baseDocPath = item.name && item.name.includes("Security Scheme Object")
+    ? 'base-doc-security-scheme-object.json'
+    : 'base-doc.json';
+
+  // Check if the directory exists, and if not, create it
+  if (!fs.existsSync(outputDir)) {
+    fs.mkdirSync(outputDir, { recursive: true });
+  }
+
+  const processedExampleName = item.name.replace(/[^\w\s-]/g, '').replace(/\s+/g, '-');
+  const outputPath = path.join(outputDir, `${processedExampleName}.json`);
+  // const outputPath = `./updated-docs/updated-doc-${index + 1}.json`;
+  // console.log(`\n${combinedData[num-1].name} = ${currentExample}`);
+
+  // Apply updates and save the document
+  applyUpdatesAndSave([item], baseDocPath, outputPath);
+
+  // Validate the output file using the AsyncAPI parser
+  // const validationPromise = validateParser(outputPath);
+
+  // Use the AsyncAPI CLI to validate the output file
+  const validationPromise = validateCli(outputPath)
```

Comment:
Parser should be enough. Bringing the CLI here is too much in terms of dependencies IMHO

Reply:
@smoya 
Okay, I'll add the validation through parser.
Thought of adding the CLI due to more verbose details of the error.

Reply:
> Thought of adding the CLI due to more verbose details of the error.

Think about the cost of calling this through command line, meaning you are spinning up new processes on the machine, waiting for the process output, trusting npx binary is installed, etc, etc VS just using the `@asyncapi/parser` package. Additionally IMHO we don't really need such a verbosity.

Reply:
> > Thought of adding the CLI due to more verbose details of the error.
> 
> Think about the cost of calling this through command line, meaning you are spinning up new processes on the machine, waiting for the process output, trusting npx binary is installed, etc, etc VS just using the `@asyncapi/parser` package. Additionally IMHO we don't really need such a verbosity.

Got it! Thanks for the detailed explanation. Very much appreciate it. :grin:

Reply:
resolved through d7eb944